src/oop/static_modifier/hard/q2/MainTest.java

```diff
@@ -0,0 +1,162 @@
+package oop.static_modifier.hard.q2;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static global.tools.CustomAssertions._assertEquals;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String className = "Lightbulb";
+    private final String staticAttributeName = "voltage";
+    private final String attributeName = "type";
+    private final String setAttributeName = "setVoltage";
+    private final String testClassName = "Test";
+
+    private ObjectTest testClass;
+    private ObjectTest classObject;
+
+    @BeforeEach
+    public void setUp() {
+        String packageString = "oop.static_modifier.hard.q2." + className;
+        testClass = new ObjectTest(packageString);
+        String testClassString = "oop.static_modifier.hard.q2." + testClassName;
+        classObject = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void lightbulbHasRequiredAttributes() {
+        String missingAttributeMessage = "The %s class is missing the %s attribute. Make sure that the class contains the attribute and it is spelt correctly.";
+        String wrongTypeMessage = "The %s attribute in the %s class has the wrong type.";
+        String wrongModifierMessage = "The %s attribute in the %s class has the wrong visibility modifier.";
+        assertTrue(testClass.hasField(attributeName), String.format(missingAttributeMessage, className, attributeName));
+        assertTrue(testClass.hasField(attributeName, String.class), String.format(wrongTypeMessage, attributeName, className));
+        assertTrue(testClass.hasModifier(attributeName, "private"), String.format(wrongModifierMessage, attributeName, className));
+        assertTrue(testClass.hasField(staticAttributeName), String.format(missingAttributeMessage, className, staticAttributeName));
+        assertTrue(testClass.hasField(staticAttributeName, int.class), String.format(wrongTypeMessage, staticAttributeName, className));
+        assertTrue(testClass.hasModifier(staticAttributeName, "private"), String.format(wrongModifierMessage, staticAttributeName, className));
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"), String.format(wrongModifierMessage, staticAttributeName, className));
```

Comment:
This error message is not accurate because "static" is not a visibility modifier